test/base.test.js

```diff
@@ -580,3 +580,24 @@ test('calls onError listener', async (t) => {
     t.equal(err.message, 'whoops')
   }
 })
+
+test('should call onError when serialize throws exception', async (t) => {
+  t.plan(2)
+
+  const serialize = () => {
+    throw new Error('error serializing')
+  }
+
+  const onError = err => t.equal(err.message, 'error serializing')
+
+  // onError in Cache options
+  const cache = new Cache({ storage: createStorage(), onError })
+  cache.define('serializeWithError', { serialize }, async k => k)
+
+  // onError in define options
+  const cache2 = new Cache({ storage: createStorage() })
```

Comment:
not necessary to test options here, implementation does not depends by options
if you want to do that, you should either create another test case or another `onError` listener, because if the test fails, it's not clear which of the 2 actions is failing